app/js/components/LabResultEntry.jsx

```diff
@@ -96,7 +96,9 @@ export class LabResultEntry extends PureComponent {
   componentWillUnmount() {
     const { dispatch } = this.props;
     const { labOrder } = this.state;
-    dispatch(updateLabOrderWithEncounter(labOrder));
+    if (!R.isEmpty(labOrder)) {
+      dispatch(updateLabOrderWithEncounter(labOrder));
+    }
```

Comment:
So "updateLabOrderWithEncounter" is dispatched both on component mount and unmount.  IIRC, I believe this is to update the lab order with any results or other information stored in the encounter.  It's triggered on page mount to fetch the results for that particular order when clicking from the list to the order details page.

I believe it calls it on unmount because clicking on "Save" goes back to the list page, and I believe this is to load any changes that were made to the order and saved.

When you reload the page, the component unmounts, but there doesn't seem to be an order present, which leads to a NPE downstream.  It seems safe to just skip the dispatch if the labOrder is empty.

Reply:
Nice, thanks for the explanation!